Design.md

```diff
@@ -276,3 +277,39 @@ There have been multiple rounds of CPU vulnerabilities (L1TF and MDS) which cann
 By default, Fedora CoreOS will configure the kernel to disable SMT on vulnerable machines. This conditional approach avoids incurring the performance cost on systems that aren't vulnerable. However, it fails to protect systems affected by undisclosed SMT vulnerabilities, and it allows future OS updates to disable SMT without notice if new vulnerabilities become known.
 
 We will document this policy and its consequences, and provide instructions for unconditionally enabling or disabling SMT for users who prefer a different policy.
+
+## Bucket Layout
+
+Originally discussed in [#189](https://github.com/coreos/fedora-coreos-tracker/issues/189).
+
+The `fcos-builds` bucket, fronted by http://builds.coreos.fedoraproject.org/ will be structured as follow:
+
+```
+/
+  prod/
+    streams/
+      stable/
+        stream.json
+        releases.json
+        builds/
+          builds.json
+          30.1234-5/
+            release.json
+            x86_64/
+              meta.json
+              commitmeta.json
+              fedora-coreos-30.8-qemu.x86_64.qcow2.gz
+              ostree-commit-object
+              ostree-commit.tar
+              ...
+            ppc64le/
+              ...
+            ...
+      testing/
+      next/
+      ...
+```
+
+The artifacts under e.g. `30.1234-5/x86_64/` come directly from [coreos-assembler](https://github.com/coreos/coreos-assembler). The `stream.json`, `release.json`, and `releases.json` are higher-level generated metadata objects. See [#98](https://github.com/coreos/fedora-coreos-tracker/issues/98) and [#207](https://github.com/coreos/fedora-coreos-tracker/pull/207) for more information about those.
+
+Note that while user-intended metadata will contain links to artifacts in the bucket, users shouldn't care about the exact layout, and we should be free to change it if needed in the future. As such, we should strongly discourage anything we don't control from relying on this structure.
```

Comment:
Folks will read this who are unfamiliar with FCOS, so let's emphasize this part a bit more.  For example:

> The stream metadata format is intended to be stable, and stream metadata objects will contain links to artifacts in the release bucket.  **Everything else about the bucket layout, including its directory structure and the formats of other metadata objects, is subject to change without notice.  Third-party tooling should not rely on this structure**, and should instead read metadata and artifact URLs directly from stream metadata at the officially documented URL.